src/components/TextField/TextField.tsx

```diff
@@ -432,7 +432,10 @@ class TextField extends React.PureComponent<CombinedProps, State> {
     // step / value has.
     const decimalPlaces = Math.max(dpl(numericValue), dpl(step));
 
-    const newValue = Math.min(max, Math.max(numericValue + steps * step, min));
+    const newValue = Math.min(
```

Comment:
It's neat that this now allow for doing number-like strings for max and mins (note max and min being strings `"1"` not the number `1`):

```
export default function Playground() {
  const [value, setValue] = useState('1');
  const handleChange = useCallback((newValue) => setValue(newValue), []);

  return (
    <Page title="Playground">
      <TextField
        label="thing"
        value={value}
        onChange={handleChange}
        type="number"
        min="1"
        max="10"
      />
    </Page>
  );
}

```

But it means typescript will no longer complain if you do some weird misconfiguration like setting a max to a string that can't be cast to a number e.g.

```
export default function Playground() {
  const [value, setValue] = useState('0');

  const handleChange = useCallback((newValue) => setValue(newValue), []);

  return (
    <Page title="Playground">
      <TextField
        label="thing"
        value={value}
        onChange={handleChange}
        type="number"
        min="1"
        max="2019-08-10"
      />
    </Page>
  );
}
```

In this case newValue ends up being NaN and we get console warnings. Can we add a check above this to ensure  that max and min can be cast to number and if not return without triggering onChange? Perhaps something like (thought about but not tested):

```diff
- if (onChange == null) {
+ if (onChange == null || isNaN(min)) || isNaN(min)) {
```

Reply:
> In this case newValue ends up being NaN and we get console warnings

That’s why I prefer this approach. Otherwise, with an early return for not a number the stepper no longer works without feedback to the implementer as to why. We could handle the messaging ourselves, but that is reinventing the wheel.

I also considered `isNan` values defaulting to `Infinity` or `-Infinity` which mimics the behavior of the html number input, but that makes it harder for an implementer to debug if they say, fat-fingered `10-` or something like this.

Reply:
Cool, that makes sense